pkg/endpoint/bpf.go

```diff
@@ -202,7 +202,7 @@ func (p *proxyPolicy) GetListener() string {
 // Only called after a new desired policy has been computed.
 // Must be called with endpoint.mutex Lock()ed.
 func (e *Endpoint) addNewRedirectsFromDesiredPolicy(ingress bool, desiredRedirects map[string]uint16, proxyWaitGroup *completion.WaitGroup) (error, revert.FinalizeFunc, revert.RevertFunc) {
-	if option.Config.DryMode || e.IsProxyDisabled() {
+	if e.isMetadata(MetadataFakeEndpoint) || e.IsProxyDisabled() {
```

Comment:
This comment applies to all the below changes - 

I would suggest adding a wrapper that checks `option.Config.DryMode` or whether an endpoint is a fake one. This way, we preserve the existing behavior.